basyx.submodelservice-core/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/value/TestMappedSubmodelElementValue.java

```diff
@@ -131,51 +161,242 @@ public void mappedGetFileValue() {
+
+		AnnotatedRelationshipElementValueMapper relationshipElementValueMapper = new AnnotatedRelationshipElementValueMapper(expected);
+
+		assertEqualsAnnotatedRelationshipElementValue(expected, relationshipElementValueMapper);
+	}
+
+	@Test
+	public void mappedSetAnnotatedRelationshipElementValue() {
+		AnnotatedRelationshipElement annotatedRelationshipElement = SubmodelServiceUtil.createAnnotatedRelationshipElementSubmodelElement();
+		
+		AnnotatedRelationshipElementValue relationshipElementValue = new AnnotatedRelationshipElementValue(referenceValue_first, referenceValue_second, valueOnlies);
+
+		setReferenceElementValue(annotatedRelationshipElement, relationshipElementValue);
+		
+		assertEqualsSetAnnotatedRelationshipElementValue(annotatedRelationshipElement);
+	}
+
+	private void assertEqualsSetAnnotatedRelationshipElementValue(
+			AnnotatedRelationshipElement annotatedRelationshipElement) {
+		assertEquals(((PropertyValue) valueOnlies.get(0).getSubmodelElementValue()).getValue(), ((Property) annotatedRelationshipElement.getAnnotations().get(0)).getValue());
+
+		assertEquals(((RangeValue) valueOnlies.get(1).getSubmodelElementValue()).getMin(), Integer.parseInt(((Range) annotatedRelationshipElement.getAnnotations().get(1)).getMin()));
+
+		assertEquals(((RangeValue) valueOnlies.get(1).getSubmodelElementValue()).getMax(), Integer.parseInt(((Range) annotatedRelationshipElement.getAnnotations().get(1)).getMax()));
+		
+		assertEquals(referenceValue_first.getKeys(), annotatedRelationshipElement.getFirst().getKeys());
+		assertEquals(referenceValue_first.getType(), annotatedRelationshipElement.getFirst().getType());
+		
+		assertEquals(referenceValue_second.getKeys(), annotatedRelationshipElement.getSecond().getKeys());
+		assertEquals(referenceValue_second.getType(), annotatedRelationshipElement.getSecond().getType());
+	}
+	
+	private void assertEqualsSetEntityValue(Entity entity) {
+		assertEquals(((PropertyValue) valueOnlies.get(0).getSubmodelElementValue()).getValue(), ((Property) entity.getStatements().get(0)).getValue());
+
+		assertEquals(((RangeValue) valueOnlies.get(1).getSubmodelElementValue()).getMin(), Integer.parseInt(((Range) entity.getStatements().get(1)).getMin()));
+
+		assertEquals(((RangeValue) valueOnlies.get(1).getSubmodelElementValue()).getMax(), Integer.parseInt(((Range) entity.getStatements().get(1)).getMax()));
+
+		assertEquals(testEntityType, entity.getEntityType());
+		
+		assertEquals(referenceValue_first.getKeys(), entity.getGlobalAssetId().getKeys());
+		
+		assertEquals(referenceValue_first.getType(), entity.getGlobalAssetId().getType());
+		
+		assertEquals(specificAssetIdValues.get(0).getName(), entity.getSpecificAssetId().getName());
+		
+		assertEquals(specificAssetIdValues.get(0).getValue(), entity.getSpecificAssetId().getValue());
+	}
+	
+	private void assertEqualsSetRelationshipElementValue(RelationshipElement relationshipElement) {
```

Comment:
These asserts should be static and take the expected values also as parameter, e.g., here assertEqualsSetRelationshipElementValue(first, second, relationshipElement).